compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java

```diff
@@ -3330,44 +3277,2087 @@ public void patch(int codePos, int relative, byte[] code) {
+            return !(size == ASIMDSize.HalfReg && eSize == ElementSize.DoubleWord);
+        }
+
+        /* Helper values/methods for encoding instructions */
+
+        private static final int ASIMDImm5Offset = 16;
+        private static final int ASIMDQBitOffset = 30;
+        private static final int ASIMDSizeOffset = 22;
+
+        private final int elemSize00 = 0b00 << ASIMDSizeOffset;
+        private final int elemSize01 = 0b01 << ASIMDSizeOffset;
+        private final int elemSize10 = 0b10 << ASIMDSizeOffset;
+        private final int elemSize11 = 0b11 << ASIMDSizeOffset;
+
+        private static int elemSizeXX(ElementSize eSize) {
+            return eSize.encoding << ASIMDSizeOffset;
+        }
+
+        private static int elemSize1X(ElementSize eSize) {
+            assert eSize == ElementSize.Word || eSize == ElementSize.DoubleWord;
+            return (0b10 | (eSize == ElementSize.DoubleWord ? 1 : 0)) << ASIMDSizeOffset;
+        }
+
+        private static int elemSize0X(ElementSize eSize) {
+            assert eSize == ElementSize.Word || eSize == ElementSize.DoubleWord;
+            return (eSize == ElementSize.DoubleWord ? 1 : 0) << ASIMDSizeOffset;
+        }
+
+        /**
+         * Sets the Q-bit if requested.
+         */
+        private static int qBit(boolean isSet) {
+            return (isSet ? 1 : 0) << ASIMDQBitOffset;
+        }
+
+        /**
+         * Sets the Q-bit if using all 128-bits.
+         */
+        private static int qBit(ASIMDSize size) {
+            return (size == ASIMDSize.FullReg ? 1 : 0) << ASIMDQBitOffset;
+        }
+
+        private void asimdAcrossLanes(ASIMDInstruction instr, ASIMDSize size, int eSizeEncoding, Register dst, Register src) {
+            int baseEncoding = 0b0_0_0_01110_00_11000_00000_10_00000_00000;
+            emitInt(instr.encoding | baseEncoding | qBit(size) | eSizeEncoding | rd(dst) | rs1(src));
+        }
+
+        private void asimd3Same(ASIMDInstruction instr, ASIMDSize size, int eSizeEncoding, Register dst, Register src1, Register src2) {
+            int baseEncoding = 0b0_0_0_01110_00_1_00000_00000_1_00000_00000;
+            emitInt(instr.encoding | baseEncoding | qBit(size) | eSizeEncoding | rd(dst) | rs1(src1) | rs2(src2));
+
+        }
+
+        private void asimdScalar3Same(ASIMDInstruction instr, int eSizeEncoding, Register dst, Register src1, Register src2) {
+            int baseEncoding = 0b01_0_11110_00_1_00000_00000_1_00000_00000;
+            emitInt(instr.encoding | baseEncoding | eSizeEncoding | rd(dst) | rs1(src1) | rs2(src2));
+        }
+
+        private void asimdCopy(ASIMDInstruction instr, boolean setQBit, ElementSize eSize, Register dst, Register src, int index) {
+            assert index >= 0 && index < ASIMDSize.FullReg.bytes() / eSize.bytes();
+            int baseEncoding = 0b0_0_0_01110000_00000_0_0000_1_00000_00000;
+            int imm5Encoding = (index << eSize.bytes() | eSize.bytes()) << ASIMDImm5Offset;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Nonsensical self computation involving a value (e.g., x & x) (SA_SELF_COMPUTATION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-nonsensical-self-computation-involving-a-field-e-g-x-x-sa-field-self-computation).
This method performs a nonsensical computation of a field with another reference to the same field (e.g., x&x or x-x). Because of the nature of the computation, this operation doesn't seem to make sense, and may indicate a typo or a logic error. Double check the computation.